abort-controller dependency problem

Hello Everyone,

I am getting started with LearnCard SDK, I just installed it using pnpm (pnpm install @learncard/core), but when trying to import { initLearnCard } from '@learncard/core', a dependency error appears: 

logger.ts:268 Uncaught Error: Cannot find module './dist/abort-controller'
    at webpackMissingModule (logger.ts:268:1)
    at ./node_modules/.pnpm/abort-controller@3.0.0/node_modules/abort-controller/polyfill.mjs (logger.ts:268:1)
    at options.factory (react refresh:6:1)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./node_modules/.pnpm/@learncard+core@7.0.1/node_modules/@learncard/core/dist/core.esm.js (index.js:91:1)
    at options.factory (react refresh:6:1)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./src/pages/Credentials/IssueCredential/index.jsx (index.jsx:33:1)

This issue could not be associated with learn-card but its dependency abort-controller (.mjs or .js). Reviewing abort-controller on npm it appears to not being updated 4 years ago.

---------------------------

App info:

npm -v: 8.19.2
node -v: 16.17.0

React app created with create-react-app, with the following dependencies:

    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-router-dom": "6.4.2",
    "firebase": "9.12.1",
    "@mui/material": "5.10.9",
    "@mui/icons-material": "^5.10.9",
    "@emotion/react": "^11.10.4",
    "@emotion/styled": "^11.10.4",

When installing via pnpm, following version of learncard is added:
    "@learncard/core": "^7.0.1",

I would appreciate if you could help me to kow how to solve this problem, or if I have made a mistake on any step or there is any missing or incompatible dependency. Thank you!

--- a/.changeset/tricky-rocks-warn.md
+++ b/.changeset/tricky-rocks-warn.md
@@ -1,0 +1,5 @@
+---
+'@learncard/core': patch
+---
+
+Fix abort-controller issues with Webpack

--- a/packages/learn-card-core/src/index.ts
+++ b/packages/learn-card-core/src/index.ts
@@ -1,8 +1,6 @@
 /// <reference path="./types/global.d.ts" />
 
-import 'core-js/actual/promise';
-import 'isomorphic-fetch';
-import 'abort-controller/polyfill';
+import './polyfills';
 
 export * from '@wallet/init';
 export { LearnCard, Plugin } from 'types/wallet';

--- a/packages/learn-card-core/src/polyfills.ts
+++ b/packages/learn-card-core/src/polyfills.ts
@@ -1,0 +1,23 @@
+import 'core-js/actual/promise';
+import 'isomorphic-fetch';
+import * as ac from 'abort-controller/dist/abort-controller';
+
+const g: any =
+    typeof globalThis !== 'undefined'
+        ? globalThis
+        : typeof self !== 'undefined'
+        ? self
+        : typeof window !== 'undefined'
+        ? window
+        : typeof global !== 'undefined'
+        ? global
+        : /* otherwise */ undefined;
+
+if (g) {
+    if (typeof g.AbortController === 'undefined') {
+        g.AbortController = ac.AbortController;
+    }
+    if (typeof g.AbortSignal === 'undefined') {
+        g.AbortSignal = ac.AbortSignal;
+    }
+}

--- a/packages/learn-card-core/src/polyfills.ts
+++ b/packages/learn-card-core/src/polyfills.ts
@@ -1,17 +1,17 @@
 import 'core-js/actual/promise';
 import 'isomorphic-fetch';
-import * as ac from 'abort-controller/dist/abort-controller';
+import * as ac from 'abort-controller/dist/abort-controller.mjs';
 
 const g: any =
     typeof globalThis !== 'undefined'
         ? globalThis
         : typeof self !== 'undefined'
-        ? self
-        : typeof window !== 'undefined'
-        ? window
-        : typeof global !== 'undefined'
-        ? global
-        : /* otherwise */ undefined;
+            ? self
+            : typeof window !== 'undefined'
+                ? window
+                : typeof global !== 'undefined'
+                    ? global
+                    : /* otherwise */ undefined;
 
 if (g) {
     if (typeof g.AbortController === 'undefined') {

--- a/packages/learn-card-core/src/polyfills.ts
+++ b/packages/learn-card-core/src/polyfills.ts
@@ -1,6 +1,6 @@
 import 'core-js/actual/promise';
 import 'isomorphic-fetch';
-import * as ac from 'abort-controller/dist/abort-controller.mjs';
+import * as ac from 'abort-controller/dist/abort-controller.js';
 
 const g: any =
     typeof globalThis !== 'undefined'

--- a/packages/learn-card-core/src/types/global.d.ts
+++ b/packages/learn-card-core/src/types/global.d.ts
@@ -27,7 +27,7 @@ declare module '@digitalbazaar/vpqr' {
     }): Promise<{ vp: VP }>;
 }
 
-declare module 'qrcode-reader' {}
+declare module 'qrcode-reader' { }
 
 declare module 'credential-handler-polyfill' {
     function loadOnce(options?: string | { mediatorOrigin?: string }): Promise<void>;
@@ -51,4 +51,9 @@ declare module 'web-credential-handler' {
         store?: (event: CredentialStoreEvent) => Promise<HandlerResponse>;
     }): void;
     function receiveCredentialEvent(): Promise<CredentialRequestEvent | CredentialStoreEvent>;
+}
+
+declare module 'abort-controller/dist/abort-controller.mjs' {
+    declare const AbortController: typeof window.AbortController;
+    declare const AbortSignal: typeof window.AbortSignal;
 }

